services/dynamodb/src/main/java/software/amazon/awssdk/services/dynamodb/model/ItemBuilder.java

```diff
@@ -0,0 +1,351 @@
+     * @return the builder for method chaining
+     */
+    public ItemBuilder attr(String key, String stringValue) {
+        item.put(key, AttributeValue.builder().s(stringValue).build());
+        return this;
+    }
+
+    /**
+     * Adds an {@link AttributeValue} representing a Boolean to the item with the specified key.
+     *
+     * Equivalent of:
+     * <pre><code>
+     * itemMap.put(key, AttributeValue.builder().bool(booleanValue).build());
+     * </code></pre>
+     *
+     * @param key the key of this attribute
+     * @param booleanValue the boolean value of the attribute
+     * @return the builder for method chaining
+     */
+    public ItemBuilder attr(String key, Boolean booleanValue) {
+        item.put(key, AttributeValue.builder().bool(booleanValue).build());
+        return this;
+    }
+
+    /**
+     * Adds an {@link AttributeValue} representing a Number to the item with the specified key.
+     *
+     * Equivalent of:
+     * <pre><code>
+     * itemMap.put(key, AttributeValue.builder().n(String.valueOf(numericValue)).build());
+     * </code></pre>
+     *
+     * @param key the key of this attribute
+     * @param numericValue the numeric value of the attribute
+     * @return the builder for method chaining
+     */
+    public ItemBuilder attr(String key, Number numericValue) {
+        item.put(key, AttributeValue.builder().n(String.valueOf(numericValue)).build());
+        return this;
+    }
+
+    /**
+     * Adds an {@link AttributeValue} representing binary data to the item with the specified key.
+     *
+     * Equivalent of:
+     * <pre><code>
+     * itemMap.put(key, AttributeValue.builder().b(ByteBuffer.wrap(binaryValue)).build());
+     * </code></pre>
+     *
+     * @param key the key of this attribute
+     * @param binaryValue the binary value of the attribute
+     * @return the builder for method chaining
+     */
+    public ItemBuilder attr(String key, byte[] binaryValue) {
+        return attr(key, ByteBuffer.wrap(binaryValue));
+    }
+
+    /**
+     * Adds an {@link AttributeValue} representing binary data to the item with the specified key.
+     *
+     * Equivalent of:
+     * <pre><code>
+     * itemMap.put(key, AttributeValue.builder().b(binaryValue).build());
+     * </code></pre>
+     *
+     * @param key the key of this attribute
+     * @param binaryValue the binary value of the attribute
+     * @return the builder for method chaining
+     */
+    public ItemBuilder attr(String key, ByteBuffer binaryValue) {
+        item.put(key, AttributeValue.builder().b(binaryValue).build());
+        return this;
+    }
+
+    /**
+     * Adds an {@link AttributeValue} representing a list of AttributeValues to the item with the specified key.
+     *
+     * This will attempt to infer the of of the {@link AttributeValue} for each given {@link Object} in the list
+     * based on the type using {@link AttributeValueHelper#attr(Object)}
```

Comment:
Can we list the supported types? (here and in our other object cases as well)

Reply:
done